File: nest/src/core/domain/model/entity.ts
/*
 * Copyright 2021 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */

import {Identity} from "./identity";

export interface Entity<T extends Entity<T>> {
    readonly id: Identity<T>;
}

File: quarkus-imperative/src/main/java/dev/nonava/techstacks/quarkus/imperative/core/domain/model/Entity.java
/*
 * Copyright 2021-2022 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */

package dev.nonava.techstacks.quarkus.imperative.core.domain.model;

public interface Entity<T extends Entity<T>> {
    Identity<T> getId();
}
